Makefile

```diff
@@ -103,6 +105,10 @@ all: lint test manager
 native-test:
 	GO111MODULE=on go test -mod vendor ./pkg/... ./apis/... -race -bench . -coverprofile cover.out
 
+.PHONY: benchmark-test
+benchmark-test:
+	go test ./pkg/... -bench . -run="^#" -count 5 > ${BENCHMARK_FILE_NAME}
```

Comment:
3 callouts here

* from my experience benchmarking `count=5` can yield statistically insignificant results; I usually use `count=10,20` (some 10 multiple), 

* the default timeout here is 10min but the GH action sets a timeout of 30 minutes; for debugging this may cause some confusion. Let's use the same timeout value in both places if that makes sense.

* I think it's best practice when benchmarking to set `GOMAXPROCS` here to decrease hardware variance. This is along the lines of Max's thoughts above. In particular for this cmd, let's set this to `GOMAXPROCS=1 go test ...` or some env var that we control but let's set it to have more consistent results across hardware.

Reply:
Ohh thanks for this input!